server/src/main/java/io/crate/metadata/upgrade/IndexTemplateUpgrader.java

```diff
@@ -114,4 +122,48 @@ private HashMap<String, IndexTemplateMetadata> archiveUnknownOrInvalidSettings(M
         }
         return upgradedTemplates;
     }
+
+    public static boolean populateColumnPositions(Map<String, Object> mapping) {
```

Comment:
Can this be shared with what's in the SchemaUpdateAction? At a glance it looks the same

Reply:
There is a small difference that SchemaUpdateAction assumes there are no duplicates.